index.bs

```diff
@@ -137,7 +182,46 @@ Coordinates</a></code> interface of the Geolocation API are the following:
 </div>
 
 To <dfn>construct a <a>Geolocation Sensor</a> object</dfn> the user agent must invoke the
-[=construct a Sensor object=] abstract operation.
+[=construct a Sensor object=] abstract operation for the {{GeolocationSensor}} interface.
+
+### GeolocationSensor.read() ### {#geolocationsensor-read}
+
+The {{GeolocationSensor/read()}} method, when called, must run the following algorithm:
+
+<div algorithm="read">
+<!-- https://dom.spec.whatwg.org/#abortcontroller-api-integration -->
+
+    : input
+    :: |options|, a {{ReadOptions}} object.
+    : output
+    :: |p|, a promise.
+
+    1. Let |p| be a new promise.
+    1. Let |options| be the first argument.
+    1. Let |signal| be the |options|' dictionary member of the same name if present, or null otherwise.
+    1. If |signal|'s <a>aborted flag</a> is set, then reject |p| with an "{{AbortError!!exception}}" {{DOMException}} and return |p|.
+    1. Let |geo| be the result of invoking <a>construct a Geolocation Sensor object</a> with |options|. If this throws a {{DOMException}}, catch it, reject |p| with that {{DOMException}}, and return |p|.
+    1. Invoke |geo|.{{Sensor/start()}}.
+    1. If |signal| is not null, then <a>add the following abort steps</a> to |signal|:
+      1. Invoke |geo|.{{Sensor/stop()}}.
+      1. Reject |p| with an "{{AbortError!!exception}}" {{DOMException}} and abort these steps.
+    1. Run these steps <a>in parallel</a>:
+      1. If <a>notify error</a> is invoked with |geo|, reject |p| with the {{DOMException}} passed as input to <a>notify error</a>.
```

Comment:
Single shot path should return sooner if connected and activated sensor has reading. There is no reason for scheduling new task. I think one line in the algo would not hurt.

`1. If <a>notify activated state</a> is invoked with |geo|, and  |geo|.hasReading is true, then <a>resolve geolocation promise</a> with |geo| and |p|.`

Reply:
> Single shot path should return sooner if connected and activated sensor has reading.

Are there any guarantees about how "fresh" that reading is supposed to be that one gets through `read()`? I can see it be an issue if `read()` would return the previous reading if there is one but wait for a fresh one if there isn't one.

Reply:
For the current algo, both paths would produce reading of the same "freshness", however, we would eliminate one task creation (+event loop spin), if promise is resolved immediately after sensor activation.

To quantify "freshness", there would need to be some sort of hint / options that is provided to the GeolocationSensor, e.g., `give me position that is no older than 5 minutes`.

@anssiko wdyt? Did you have a plan to add such feature? It is quite important for GeolocationSensor.